lisa/tools/sysbench.py

```diff
@@ -0,0 +1,563 @@
+    def can_install(self) -> bool:
+        return True
+
+    def install(self) -> bool:
+        posix_os: Posix = cast(Posix, self.node.os)
+        posix_os.install_packages("sysbench")
+
+        return self._check_exists()
+
+    def __run_sysbench_test(
+        self,
+        test_type: str,
+        threads: int,
+        events: int,
+        verbosity: int,
+        time_limit: int,
+        percentile: int,
+        debug: str,
+        histogram: str,
+        validation: str,
+        extra_args: str = "",
+    ) -> ExecutableResult:
+        args = (
+            f"{test_type} run --threads={threads} --events={events}"
+            f" --validate={validation} --percentile={percentile}"
+            f" --verbosity={verbosity} --histogram={histogram}"
+            f" --time={time_limit} --debug={debug} {extra_args}"
+        )
+
+        result: ExecutableResult = self.run(args)
+        return result
+
+    def run_cpu_perf(
+        self,
+        test_result: Any,
+        subtest_name: str = "",
+        threads: int = 1,
+        events: int = 0,
+        cpu_max_prime: int = 10000,
+        verbosity: int = 5,
+        time_limit: int = 10,
+        percentile: int = 95,
+        debug: str = "on",
+        histogram: str = "on",
+        validation: str = "on",
+    ) -> SysbenchTestResult:
+        extra_args: str = f" --cpu-max-prime={cpu_max_prime}"
+        result: Dict[Any, Any] = {}
+        message: str = ""
+        status = TestStatus.QUEUED
+
+        if not subtest_name:
+            subtest_name = "default_sysbench_cpu_perf_test"
+
+        res = self.__run_sysbench_test(
+            test_type="cpu",
+            threads=threads,
+            events=events,
+            verbosity=verbosity,
+            time_limit=time_limit,
+            percentile=percentile,
+            debug=debug,
+            histogram=histogram,
+            validation=validation,
+            extra_args=extra_args,
+        )
+
+        if res.exit_code == 0:
+            result = self.__process_cpu_perf_result(res.stdout)
+            message = json.dumps(result)
+            status = TestStatus.PASSED
+        else:
+            status = TestStatus.FAILED
+            message = f"{subtest_name} failed with exit code {res.exit_code}"
+
+        self.__send_subtest_msg(
+            test_result=test_result,
+            test_name=subtest_name,
+            test_status=status,
+            test_message=message,
+        )
+
+        sysbench_result = SysbenchTestResult()
+        sysbench_result.exit_code = res.exit_code
+        sysbench_result.result = result
+
+        return sysbench_result
+
+    def run_fileio_perf(
+        self,
+        test_mode: str,
+        ops: str,
+        test_result: Any,
+        subtest_name: str = "",
+        threads: int = 1,
+        events: int = 0,
+        verbosity: int = 5,
+        time_limit: int = 10,
+        percentile: int = 95,
+        debug: str = "on",
+        histogram: str = "on",
```

Comment:
The "on/off" should be wrapped as bool to avoid typo issues.

Reply:
sure

Reply:
Made some common params always on and removed from fn param.

Use boolean for on/off as suggested

Reply:
Done